Runar/Runar/Services/Memory/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  CoreDataManager.swift
+//  Runar
+//
+//  Created by Alexey Poletaev on 23.09.2022.
+//
+
+import Foundation
+import CoreData
+import UIKit
+
+final class CoreDataManager {
+
+    // MARK: - Properties
+    static let shared = CoreDataManager()
+    private let nameCoreData = "CoreData"
+    var generatorIsLoaded: Bool = false
+    var libraryIsLoaded: Bool = false
+    private var context = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
```

Comment:
Why do you use private queue?

Reply:
And why for did you create context?

Reply:
To work with Core Data on background thread, child context is required